R/fotmob_stats.R

```diff
@@ -0,0 +1,215 @@
+    league_id,
+    season_id,
+    stat_type_df$full_stat
+  )
+
+  ## This still print out HTTP error 403 even with `quiet = TRUE`?!?
+  fs <- purrr::safely(jsonlite::fromJSON, otherwise = NULL, quiet = TRUE)
+  resp <- fs(url)
+  if(!is.null(resp$error)) {
+    warning(
+      sprintf(
+        'Issue with data at `url = "%s".\n%s', url, resp$error
+      )
+    )
+    return(tibble::tibble())
+  }
+
+  # httr::stop_for_status(resp)
+  resp$result %>%
+    tibble::as_tibble() %>%
+    dplyr::select(.data$TopLists) %>%
+    tidyr::unnest(.data$TopLists) %>%
+    dplyr::select(.data$StatList) %>%
+    tidyr::unnest(.data$StatList) %>%
+    janitor::clean_names()
+}
+
+#' Get season statistics from fotmob
+#'
+#' Returns team or player season-long statistics standings from fotmob.com.
+#'
+#' @inheritParams fotmob_get_league_matches
+#' @param season_name Season names in the format `"2021/2022"`. Multiple allowed. If multiple leagues are specified, season stats are retrieved for each league.
+#'
+#' Note that not Fotmob currently only goes back as far as `"2016/2017"`. Some leagues may not have data for that far back.
+#'
+#' @param team_or_player return statistics for either \code{"team"} or \code{"player"}. Can only be one or the other.
+#' @param stat_type the type of statistic.
+#' For \code{entity = "player"}, must be one of the following:
+#' \itemize{
+#' \item{"accurate_pass"}
+#' \item{"big_chance_created"}
+#' \item{"big_chance_missed"}
+#' \item{"clean_sheet"}
+#' \item{"effective_clearance"}
+#' \item{"xg"}
+#' \item{"xg_conceded"}
+#' \item{"assist"}
+#' \item{"goals"}
+#' \item{"mins_played_goal"}
+#' \item{"ontarget_scoring_att"}
+#' \item{"penalty_won"}
+#' \item{"poss_won_att_3rd"}
+#' \item{"rating"}
+#' \item{"red_card"}
+#' \item{"saves"}
+#' \item{"total_att_assist"}
+#' \item{"won_contest"}
+#' \item{"won_tackle"}
+#' \item{"yellow_card"}
+#' }
+#' For `entity = "team"` all of the same stats are available, with the exception of:
+#' \itemize{
+#' \item{"goals"}
+#' \item{"assist"}
+#' }
+#' Additional stats available for \code{"team"} are:
+#' \itemize{
+#' \item{"goals_conceded_per_match"}
+#' \item{"goals_per_match"}
+#' \item{"possession_percentage"}
+#' }
+#'
+#' @return returns a dataframe of league standings
+#'
+#' @importFrom purrr possibly pmap_dfr
+#' @importFrom tibble tibble
+#' @importFrom rlang maybe_missing
+#' @importFrom dplyr filter distinct pull
+#'
+#' @export
+#' @examples
+#' \dontrun{
+#' library(dplyr)
+#' library(tidyr)
+#' }
+fotmob_get_season_stats <- function(
+  country,
+  league_name,
+  league_id,
+  season_name,
+  team_or_player,
+  stat_type
+) {
+
+  ## There's already warnings in this function for non-matching leagues, so we don't need to warn about that here.
+  urls <- .fotmob_get_season_ids(
+    country = rlang::maybe_missing(country, NULL),
+    league_name = rlang::maybe_missing(league_name, NULL),
+    league_id = rlang::maybe_missing(league_id, NULL)
+  )
```

Comment:
similar set-up to the `fotmob_get_league_matches` and `fotmob_get_league_tables` functions